mongo/collection.go

```diff
@@ -388,7 +398,14 @@ func (coll *Collection) InsertOne(ctx context.Context, document any,
 	}
 	if rawDataOpt := optionsutil.Value(args.Internal, "rawData"); rawDataOpt != nil {
 		imOpts.Opts = append(imOpts.Opts, func(opts *options.InsertManyOptions) error {
-			optionsutil.WithValue(opts.Internal, "rawData", rawDataOpt)
+			opts.Internal = optionsutil.WithValue(opts.Internal, "rawData", rawDataOpt)
+
+			return nil
+		})
+	}
+	if rawDataOpt := optionsutil.Value(args.Internal, "addCommandFields"); rawDataOpt != nil {
+		imOpts.Opts = append(imOpts.Opts, func(opts *options.InsertManyOptions) error {
+			opts.Internal = optionsutil.WithValue(opts.Internal, "addCommandFields", rawDataOpt)
```

Comment:
`rawDataOpt` is confusingly named because we're copying the "addCommandFields" option here. Rename to something like `ac` or `acOpt`.


```suggestion
	if ac := optionsutil.Value(args.Internal, "addCommandFields"); ac != nil {
		imOpts.Opts = append(imOpts.Opts, func(opts *options.InsertManyOptions) error {
			opts.Internal = optionsutil.WithValue(opts.Internal, "addCommandFields", ac)
```